tuf/signed/verify.go

```diff
@@ -107,3 +108,18 @@ func VerifySignature(msg []byte, sig *data.Signature, pk data.PublicKey) error {
 	sig.IsValid = true
 	return nil
 }
+
+// VerifyPublicKeyMatchesPrivateKey checks if the private key and the public keys forms valid key pairs.
+// This should work with both ecdsa-x509 certificate PublicKey as well as ecdsa PublicKey
+func VerifyPublicKeyMatchesPrivateKey(privKey data.PrivateKey, pubKey data.PublicKey) error {
+	pubKeyID, err := utils.CanonicalKeyID(pubKey)
+	privKeyID := privKey.ID()
```

Comment:
Before we call `privKey.ID()`, do we want to check to make sure it's not nil?  (maybe we can just do so in the if block below:  `if privKey == nil || pubKeyID != privKey.ID {...`)

Likewise, in the tests (thanks for adding those!), particularly in `TestVerifyPublicKeyMatchesPrivateKeyFails`, could we also test passing a nil public key and/or nil private key?